### DISCUSSION TITLE
Find & set (other) source for installed applications?

### DISCUSSION THREAD

**Discussion by JustMyGithub** [Devolutions/UniGetUI#2651] [Q&A] [answered]
I'm new to UniGetUI, so I hope I did not miss that feature in Github Issues, Github Discussions and the GUI:
How can I look for other sources of installed applications? Either because it has no repo source yet (and therby cannot be updated, I guess?) or because I want to find out whether it is available on another repo that is quicker in updating to new versions.

**Comment by Lockszmith-GH** [comment, marked as answer]
Short anwers is - you can't.

UniGetUI uses the underlying engines, and brings it to a single UI to pick and choose from.
If there is a way to do that with the commadn line tools, it might be possible to add a feature based on those capabilities.

Is this a feature of one of the cli sources you're looking to mimic in the UI?

**Reply by JustMyGithub** [reply]
Like winget saying "oh by the way, this is also an chocolaty with uri 123456 and on scoop with uri 654321"?
If this is a meta-UI for those repo-types it is the UIs task to implement cross-repo-type-features, isn't it?
It may be as easy as searching for the name on all relevant supported repos (obviously you will not search on npm and pip, as they are not for apps but for libraries of a certain language)

Also it is kind of weird that "Universal Get" cannot get updates for the most common type of software: manually installed.
Think of users that never used winget/Scoop/chocolaty before. How awesome it would be to show them "we can update all these tools for you".

**Reply by Lockszmith-GH** [reply]
As long as you have those repos on, you'll get the info.

If I'm not using those repos, why should I want to know it's available there?

**Reply by Lockszmith-GH** [reply]
You're thinking in marketing terms, I'm thinking in engieering terms.
I want to use what I know about.

I've enabled everyting at the first time, and immedietly disabled everything I'm not using, as it was too noisy

I would rather go with the educational course, of letting them know through a tool like UniGetUI, that other options exist should they want to explore, but not on the package search, I feel that would be TMI.

**Reply by JustMyGithub** [reply]
> As long as you have those repos on, you'll get the info.
> If I'm not using those repos, why should I want to know it's available there?

No. If I explicitly search for Total Commander I am shown its available on chocolaty and winget. In "software updates" and "installed packages" it does show winget only (as I installed it using winget) and does not show availability on chocolaty. So "As long as you have those repos on, you'll get the info" may be true for the search section, but not in general. Choco is on, but the installed list does not show it.

**Reply by Lockszmith-GH** [reply]
You mean like this #2598 FR/Idea?
